std/format.d

```diff
@@ -1469,10 +1469,30 @@ if (is(IntegralTypeOf!T) && !is(T == enum) && !hasToString!(T, Char))
 
     // Forward on to formatIntegral to handle both U and const(U)
     // Saves duplication of code for both versions.
-    static if (isSigned!U)
-        formatIntegral(w, cast( long) val, f, base, Unsigned!U.max);
+    static if (__traits(compiles, { ucent uc; }))
```

Comment:
It might be possible to use `is(ucent == U)` to detect `ucent` support and check for a match in the same expression.